### DISCUSSION TITLE
Is there any way to disable complete chain verification

### DISCUSSION THREAD

**Discussion by Jaditya-89** [strongswan/strongswan#1906] [General]
Hi , 
I have strongswan config on my nodes and auth method is pubkey . 

On peer 1 : ID1 -> IssuingCA1 -> Intermediary CA -> Root CA 
On peer 2 : ID2 -> IssuingCA2 -> Intermediary CA -> Root CA 

Due to security purposes, I have replaced subject names to dummy values

I have 2 cases : 

1. When root CA is present in x509ca on both peers
2. When root CA is not present in x509ca folder on both peers

Logs for above cases
1. With Using root CA :
07[CFG]   using certificate "ID1"
07[CFG]   using trusted intermediate ca certificate "IssuingCA1"
07[CFG]   using trusted intermediate ca certificate "IntermediaryCA"
07[CFG]   using trusted ca certificate "Root CA"
07[CFG]   reached self-signed root ca with a path length of 2

2. Without using root CA : 
10[CFG]   using certificate "ID1"
10[CFG]   using trusted intermediate ca certificate "IssuingCA1"
10[CFG]   using trusted intermediate ca certificate "IntermediaryCA"
10[CFG] no issuer certificate found for "IntermediaryCA"
10[CFG]   issuer is "Root CA"
10[IKE] no trusted RSA public key found for 'ID1l'
10[ENC] generating IKE_AUTH response 1 [ N(AUTH_FAILED) ]

My question is ? As the Intermediary CAs are same for both peers, why I could not establish session and need whole CA chain to establish session ? 

Also, is there any way to disable this complete chain validation in strongswan/swanctl.conf ?

Thanks in advance

**Comment by tobiasbrunner** [comment]
> Also, is there any way to disable this complete chain validation in strongswan/swanctl.conf ?

No.
